libraries/SITL/SIM_Airspeed_DLVR.cpp

```diff
@@ -0,0 +1,65 @@
+#include "SIM_Airspeed_DLVR.h"
+
+int SITL::Airspeed_DLVR::rdwr(I2C::i2c_rdwr_ioctl_data *&data)
+{
+    struct I2C::i2c_msg &msg = data->msgs[0];
+    if (msg.flags == I2C_M_RD) {
+        // driver is attempting to receive reading...
+        if (msg.len != 4) {
+            AP_HAL::panic("Unexpected message length (%u)", msg.len);
+        }
+
+        union PACKED {
+            uint32_t bytes;
+            struct {
+                uint32_t filler : 5;
```

Comment:
We should avoid structures like this for representing anything serialized, as it appears to be a compiler option for how to pack and handle this. See [here](https://en.cppreference.com/w/cpp/language/bit_field) for more details (or to tell me I've misunderstood).

As a note using this stuff for an internal reason to save space is still fine and safe, it's just not something that should be used for serialization anywhere.

Reply:
This structure will work fine on all MCUs we support. On MCUs that won't get this right there are plenty of other parts of ArduPilot that will be in trouble. So I'm fine with using this approach

Reply:
I should say though, that I'm not a fan of structure names like "stuff"

Reply:
So a couple points that are salient on it:
- It's SITL where we have the most variety of compiler versions we can't control
- Because it's compiler dependent it may suddenly break for other reasons. It probably won't as it's a common thing. But it's a silly thing to lose the sim compatability for random people, where as the method used in the actual DLVR driver is robust against that.
- The order here is a bit surprising actually. The status bits are actually the high bits, and looking at the structure it looks like they are actually the low bits?

Reply:
I've changed this out for traditional bitshift.  I think we learnt something with the exercise, 'though!

And removed `stuff` - geeze, picky....